services/web/server/src/simcore_service_webserver/folders/_folders_api.py

```diff
@@ -1,67 +1,88 @@
 # pylint: disable=unused-argument
 
 import logging
-from typing import cast
 
 from aiohttp import web
-from aiopg.sa.engine import Engine
-from models_library.api_schemas_webserver.folders import FolderGet, FolderGetPage
+from models_library.api_schemas_webserver.folders_v2 import FolderGet, FolderGetPage
 from models_library.folders import FolderID
 from models_library.products import ProductName
-from models_library.projects_access import AccessRights
 from models_library.rest_ordering import OrderBy
-from models_library.users import GroupID, UserID
-from pydantic import NonNegativeInt, parse_obj_as
-from simcore_postgres_database import utils_folders as folders_db
+from models_library.users import UserID
+from models_library.workspaces import WorkspaceID
+from pydantic import NonNegativeInt
 
-from .._constants import APP_DB_ENGINE_KEY
 from ..users.api import get_user
+from ..workspaces.api import get_workspace
+from ..workspaces.errors import WorkspaceAccessForbiddenError
+from . import _folders_db as folders_db
 
 _logger = logging.getLogger(__name__)
 
 
 async def create_folder(
     app: web.Application,
     user_id: UserID,
-    folder_name: str,
-    description: str | None,
+    name: str,
     parent_folder_id: FolderID | None,
     product_name: ProductName,
+    workspace_id: WorkspaceID | None,
 ) -> FolderGet:
     user = await get_user(app, user_id=user_id)
 
-    engine: Engine = app[APP_DB_ENGINE_KEY]
-    async with engine.acquire() as connection:
-        # NOTE: folder permissions are checked inside the function
-        folder_id = await folders_db.folder_create(
-            connection,
-            product_name=product_name,
-            name=folder_name,
-            gids={user["primary_gid"]},
-            description=description if description else "",
-            parent=parent_folder_id,
+    _private_workspace_user_id: UserID | None = user_id
```

Comment:
how can it be none? above it says it is a UserID

Reply:
I improved naming: `_personal_workspace_user_id_or_none` basically that is how you differentiate between personal workspace and shared workspace. If its personal it is bound to the user id.